File: 551. Student Attendance Record I/551.go
package leetcode

import(
	"strings"
)

func checkRecord(s string) bool {
	a, l := 0, 0
    for _, v := range s{
		if v == 'A'{
			a++
		}
		if v == 'L'{
			l++
		}else{
			l=0
		}

		if a>1 || l>2{
			return false
		}
	}
	return true
}

func checkRecord1(s string) bool {
    return !(strings.Count(s, `A`) > 1) && !strings.Contains(s, `LLL`)
}

File: 551. Student Attendance Record I/_551.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        l,a = 0,0
        for i in s:
            if i == 'A':
                a+=1
            if i == 'L':
                l+=1
            else:
                l=0
            if a>1 or l>2:
                return False
        return True

    def checkRecord1(self, s):
        return s.count('A') <=1 and 'LLL' not in s